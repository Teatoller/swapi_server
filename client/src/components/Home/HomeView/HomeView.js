import React from "react";
import SwapiMenu from "../../common/SwapiMenu";

export const HomeView = ({ stars, searchSpace }) => {
  return (
    <>
      <div className="row">
        <div className="col-sm">
          <h4 className="text-secondary m-0">Star Wars People</h4>
          <div className="mb-1">
            <SwapiMenu searchSpace={searchSpace} />
          </div>

          <div className="table-responsive">
            <table className="table table-sm table-md table-striped table-hover table-bordered text-secondary">
              <thead>
                <tr>
                  <th>#</th>
                  <th scope="col">Name</th>
                  <th scope="col">Height</th>
                  <th scope="col">Mass</th>
                  <th scope="col">Gender</th>
                  <th scope="col">Homeworld</th>
                  <th scope="col">Action</th>
                </tr>
              </thead>
              <tbody>{stars}</tbody>
            </table>
          </div>
        </div>
      </div>
    </>
  );
};
export default HomeView;
