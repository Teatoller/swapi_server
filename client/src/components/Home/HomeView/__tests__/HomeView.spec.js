import React from "react";
import renderer from "react-test-renderer";
import {HomeView} from "../HomeView";

describe("HomeView", () => {
  it("snapshot renders correctly", () => {
    const component = renderer.create(<HomeView />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
