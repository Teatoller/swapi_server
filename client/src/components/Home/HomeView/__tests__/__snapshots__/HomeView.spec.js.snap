// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView snapshot renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-sm"
  >
    <h6
      className="text-secondary m-0"
    >
      Star Wars People
    </h6>
    <div
      className="mb-1"
    >
      <div
        className="d-inline p-2 bg-light text-white"
      >
        <input
          className="border border-info rounded w-25 text-muted text-center"
          onChange={[Function]}
          placeholder="Search"
          type="text"
        />
      </div>
    </div>
    <div
      className="table-responsive"
    >
      <table
        className="table table-sm table-md table-striped table-hover table-bordered text-secondary mb-0"
      >
        <thead>
          <tr>
            <th>
              #
            </th>
            <th
              scope="col"
            >
              Name
            </th>
            <th
              scope="col"
            >
              Height
            </th>
            <th
              scope="col"
            >
              Mass
            </th>
            <th
              scope="col"
            >
              Gender
            </th>
            <th
              scope="col"
            >
              Homeworld
            </th>
            <th
              scope="col"
            >
              Action
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
